plugins/modules/net_tools/nsupdate.py

```diff
@@ -427,7 +427,7 @@ def ttl_changed(self):
         if lookup.rcode() != dns.rcode.NOERROR:
             self.module.fail_json(msg='Failed to lookup TTL of existing matching record.')
 
-        current_ttl = lookup.answer[0].ttl
+        current_ttl = lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl
```

Comment:
I think this fallback should only happen for NS records, since `lookup.authority` always contains NS entries. Maybe
```suggestion
        if self.module.params['type'] == 'NS:
            current_ttl = lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl
        else:
            current_ttl = lookup.answer[0].ttl
```